sources/engine/Stride.Graphics/Vulkan/PipelineState.Vulkan.cs

```diff
@@ -158,22 +157,29 @@ private unsafe void Recreate()
                 viewportCount = 1,
             };
 
-            fixed (VkDynamicState* dynamicStatesPointer = &dynamicStates[0])
+            // TODO: Verify `(array.Length == 0) == (fixed (T* f = array) f == null)`;
```

Comment:
Yes, its obvious many people don't know this. The conditionals in the Vulkan code are a prime example. Its often hidden because its accompanied by `array.Length` or similar, but it could be causing some of the access violations.